subprojects/execution/src/main/java/org/gradle/internal/execution/steps/CatchExceptionStep.java

```diff
@@ -34,8 +33,7 @@ public CatchExceptionStep(Step<C, ? extends Result> delegate) {
     public Result execute(C context) {
         try {
             return delegate.execute(context);
-        } catch (Throwable t) {
-            ExecutionException failure = new ExecutionException(context.getWork(), t);
```

Comment:
I was wondering if we need to add some context [here](https://github.com/gradle/gradle/blob/f918d2f94cc822a7866e7f033bed94a8204d4cad/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformationStep.java#L107), though I think the context set up here may already enough: https://github.com/gradle/gradle/blob/a3e6370fd957fea55413a4838ac688865b168a33/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/TransformingArtifactVisitor.java#L61

It  does not include the class name of the transform action which failed, though :(

WDYT?